snooty/parser.py

```diff
@@ -389,14 +392,24 @@ def handle_directive(
                 ):
                     pass
                 else:
-                    msg = f'"{name}" could not open "{argument_text}: No such file exists"'
+                    msg = f'"{name}" could not open "{argument_text}": No such file exists'
                     self.diagnostics.append(Diagnostic.error(msg, util.get_line(node)))
 
         if options:
             doc["options"] = options
 
         doc["children"] = []
 
+    def validate_doc_role(self, node: docutils.nodes.Node) -> None:
+        """Validate target for doc role"""
+        fileid, target_path = util.reroot_path(
+            Path(node["target"] + ".txt"), self.docpath, self.source_path
+        )
+
+        if not target_path.is_file():
+            msg = f'"{node["name"]}" could not open "{node["target"] + ".txt"}": No such file exists'
```

Comment:
Compute `node["target"] + ".txt"` in just one place in this function, rather than twice